chore: Update DEV_SETUP.md to change recommended python versions

alexjoel42 opened the pull request:
<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview

3.7.15 came out and we recommend 3.7.13. 

Also added some quick Pyenv instructions. 

<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Test Plan

<!--
Use this section to describe the steps that you took to test your Pull Request.
If you did not perform any testing provide justification why.

OT-3 Developers: You should default to testing on actual physical hardware.
Once again, if you did not perform testing against hardware, justify why.

Note: It can be helpful to write a test plan before doing development

Example Test Plan (HTTP API Change)

- Verified that new optional argument `dance-party` causes the robot to flash its lights, move the pipettes,
then home.
- Verified that when you omit the `dance-party` option the robot homes normally
- Added protocol that uses `dance-party` argument to G-Code Testing Suite
- Ran protocol that did not use `dance-party` argument and everything was successful
- Added unit tests to validate that changes to pydantic model are correct

-->

# Changelog
change all instances of 3.7.13 to 3.7.15. 


Then I couldn't get Pyenv working so I added how to use pye

<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests

Unclear.


<!--
Describe any requests for your reviewers here.
-->

# Risk assessment

Please check if 3.7.13 causes errors for other folks before updating. It could be specific to my system setup because it's a bit quirky. 


<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

alexjoel42 committed:
Update DEV_SETUP.md

I couldn't get it working with older Python version 3.13. I just got so many SSL errors. 

WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/aenum/

Then I couldn't get Pyenv working so I added how to use pye

koji reviewed (approved):
Thank you for updating this!
tested `make setup` with python 3.7.15 and it worked as expected

@alexjoel42 
I ran the following command to pass the CI check.

```shell
npx prettier --write DEV_SETUP.md
```

koji committed:
fix linting issue

ecormany commented on DEV_SETUP.md:
```diff
@@ -199,13 +199,14 @@ Once you are inside the repository for the first time, you should do two things:
 
 1. Confirm that `nvs` selected the proper version of Node.js to use
 2. Tell `pyenv` to use Python 3.7
+3. If you aren't seeing `python --version` for an M1 Mac outputing the right value, try running `pyenv init` then run `pyenv local 3.7.15` and see if it works this time when you select `python -version`.
```
```suggestion
3. Run `python --version` to confirm your chosen version. If you get the incorrect version and you're using an Apple silicon Mac, try running `pyenv init` and then `pyenv local 3.7.15`. Then check `python --version` again.
```
You don't have to take this whole suggestion, but at least get the double dash flag on `python --version` at the very end. `python -version` will produce an error.

"Apple silicon" is to future-proof. We already have M1 and M2 and we can't list all the Ms forever into the future.

ecormany reviewed (changes requested):
Spotted a syntax error we ought to fix before merging. All the other prose suggestions are optional.

alexjoel42 committed:
Update DEV_SETUP.md

Co-authored-by: Ed Cormany <ecormany@gmail.com>

alexjoel42 committed:
Update DEV_SETUP.md

I learned that pyenv updated  in 2022 https://stackoverflow.com/questions/33321312/cannot-switch-python-with-pyenv

alexjoel42 commented:
I added a small thing to fix my [ 'pyenv init' ](https://stackoverflow.com/questions/33321312/cannot-switch-python-with-pyenv) error. 

Somehow when I PR I feel like I cause problems while trying to fix them haha.

koji commented:
@alexjoel42 
We use [commitizen](https://github.com/commitizen/cz-cli)

For the next time please follow ↓
https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#making-your-commit

If you have any question on opening a PR, feel free to ask questions #software-team channel in Slack.
